### DISCUSSION TITLE
file size should be shown in Kb/Mb/... instead of raw bytes

### DISCUSSION THREAD

**Discussion by briveira** [directus/directus#16215] [Archive]
when the user views a file, size is presented in raw form ("37227459")

![image](https://user-images.githubusercontent.com/1811641/198385253-be13529c-12bf-47bf-a3ff-1c5cc1ee3226.png)

how could we change this presentation to the one used for example in the file library?

![image](https://user-images.githubusercontent.com/1811641/198385544-f1049401-16da-469b-b030-dc7a4a4fed3f.png)

...which is more "human readable" :) "996.0 kB"

thanks in advance!

**Comment by azrikahar** [comment]
This was also brought up in #4690, but as https://github.com/directus/directus/discussions/4690#discussioncomment-534444 mentioned, there is no filesize interface at the moment. However it could be a potential improvement 👍

**Comment by briveira** [comment]
That comment says "only a display" in v9... but that's exactly what I need:  displaying, not editing (as this comes from the file library, we are not looking for editing the value, only displaying it in human readable form)

**Reply by azrikahar** [reply]
Unfortunately in the context of Directus (or at least v9), the greyed-out fields in forms in general (such as the fields in File items in your screenshot) are actually [interfaces](https://docs.directus.io/extensions/interfaces.html), not [displays](https://docs.directus.io/extensions/displays.html). 

Totally understand that in this scenario it looks like it is "displaying" the file size value which may cause confusion, but technically it is currently a read-only interface, which is why the linked comment points out there's no filesize interface yet 👍

Just to elaborate more on the difference, here we can observe the file size field is actually an [`input` interface](https://github.com/directus/directus/tree/main/app/src/interfaces/input):

![](https://user-images.githubusercontent.com/42867097/198562958-263e841a-06a9-4d82-815c-df01331a1ee9.png)

Whereas in the Cards view, it is the [`render-template` component](https://github.com/directus/directus/blob/main/app/src/views/private/components/render-template.vue), which then nests the [`filesize` display](https://github.com/directus/directus/tree/main/app/src/displays/filesize) (however since render-template uses the vue `<component>` dynamic component, we are only seeing `<Component>` instead of `<Filesize>` in the devtools!):

![](https://user-images.githubusercontent.com/42867097/198563170-43e67678-ce8f-4ae6-b99f-6ea9168bbd1b.png)

Hope that helps to clarify the distinction between them 😊

**Reply by briveira** [reply]
Yes, thank you very much!
